editoast/src/views/infra/lines.rs

```diff
@@ -0,0 +1,57 @@
+use crate::error::Result;
+use crate::infra::Infra;
+use crate::infra_cache::{InfraCache, ObjectCache};
+use crate::map::BoundingBox;
+
+use crate::DbPool;
+use actix_web::dev::HttpServiceFactory;
+use actix_web::get;
+use actix_web::web::{scope, Data, Json, Path};
+use chashmap::CHashMap;
+use diesel::sql_types::{BigInt, Bool};
+use diesel::QueryDsl;
+use diesel::{ExpressionMethods, RunQueryDsl};
+use serde::Serialize;
+
+/// Return `/infra/<infra_id>/lines` routes
+pub fn routes() -> impl HttpServiceFactory {
+    scope("/lines").service(get_line_bbox)
+}
+
+#[derive(Debug, Serialize)]
+pub struct BboxResponse {
+    geo: Vec<BoundingBox>,
+    sch: Vec<BoundingBox>,
+}
+
+#[get("/{line_code}/bbox")]
+pub async fn get_line_bbox(
+    path: Path<(i64, i64)>,
+    infra_caches: Data<CHashMap<i64, InfraCache>>,
+    db_pool: Data<DbPool>,
+) -> Result<Json<BboxResponse>> {
+    use crate::tables::osrd_infra_tracksectionmodel::dsl;
+    let (infra_id, line_code) = path.into_inner();
+    let conn = &mut db_pool.get().expect("Failed to get DB connection");
+    let ts_ids: Vec<String> = dsl::osrd_infra_tracksectionmodel
+        .filter(dsl::infra_id.eq(infra_id))
+        .filter(
+            diesel::dsl::sql::<Bool>("(\"data\"#>>'{extensions,sncf,line_code}')::integer = ")
+                .bind::<BigInt, _>(line_code),
+        )
+        .select(dsl::obj_id)
+        .get_results(conn)?;
+    let infra = Infra::retrieve(conn, infra_id)?;
+    let infra_cache = InfraCache::get_or_load(conn, &infra_caches, &infra)?;
+    let track_sections = infra_cache.track_sections();
+    let mut geo = vec![];
+    let mut sch = vec![];
+    for id in ts_ids {
+        let Some(ObjectCache::TrackSection(track)) = track_sections.get(&id) else {
+            panic!("line {line_code} (infra_id = {infra_id}) contains track section {id} but it could not be found in the infra cache");
+        };
```

Comment:
```suggestion
        let track = track_sections.get(&id).expect("line {line_code} (infra_id = {infra_id}) contains track section {id} but it could not be found in the infra cache");
```